### PR TITLE
Use local plugins, not from hub for tests

### PR DESCRIPTION
**Pull request by cpuguy83**
Use the (new) plugin fixtures for plugin tests rather than pulling
plugins from hub.

This removes the restriction for platforms/archs since plugin binaries
get built in the test environment.

Future work would be to add test plugins for the various subsystems so
tests that are actually using plugins (e.g. volumes, networks) can be
ported to use the fixtures as well.

### CODE REVIEW

**Comment by aaronlehmann** [comment]
There's a static analysis warning of some sort:

```
20:44:23 integration-cli/trust_server_test.go:245: the cancel function is not used on all paths (possible context leak)
20:44:23 integration-cli/trust_server_test.go:248: this return statement may be reached without using the cancel var defined on line 245
```

**Comment by aaronlehmann** [comment]
This is great.

Afterwards, I think we should remove `DockerHubPullSuite`. This doesn't provide much value over testing with a local registry, and has been a long-time source of flakiness.

**Comment by tophj-ibm** [comment]
Looking at the logs, it seems all the tests are timing out because of a lingering volume plugin which which is adding 15 seconds per test that comes after it. 

Looking at the time deltas between times the test took, and times they reported taking, I think the volume plugin created in DockerSwarmSuite.TestSwarmVolumePlugin is the culprit here.

**Comment by cpuguy83** [comment]
Why are these tests being reported as in `DockerPluginSuite`?

**Comment by cpuguy83** [comment]
> Afterwards, I think we should remove DockerHubPullSuite

Agreed it would make things better on CI in this repo, but I do think there is value in it (IIRC, these were added after some really bad issue). Perhaps moving these to a nightly somewhere (outside moby?) would suffice.

**Comment by cpuguy83** [comment]
Windows error:
> 10:53:32 exec: "gcc": executable file not found in %PATH%

**Comment by cpuguy83** [comment]
janky failure:

```
10:56:26 FAIL: check_test.go:360: DockerSwarmSuite.TearDownTest
10:56:26 
10:56:26 unmount of /tmp/docker-execroot/dae782a29f5c7/netns failed: invalid argument
10:56:26 unmount of /tmp/docker-execroot/dae782a29f5c7/netns failed: no such file or directory
10:56:26 Daemon defcdfc1ba008 is not started
10:56:26 unmount of /tmp/docker-execroot/defcdfc1ba008/netns failed: invalid argument
10:56:26 unmount of /tmp/docker-execroot/defcdfc1ba008/netns failed: no such file or directory
10:56:26 check_test.go:365:
10:56:26     d.Stop(c)
10:56:26 daemon/daemon.go:392:
10:56:26     t.Fatalf("Error while stopping the daemon %s : %v", d.id, err)
10:56:26 ... Error: Error while stopping the daemon dd4f748b5d563 : exit status 130
10:56:26 
10:56:26 
10:56:26 ----------------------------------------------------------------------
10:56:26 PANIC: docker_cli_swarm_test.go:1146: DockerSwarmSuite.TestSwarmLockUnlockCluster
10:56:26 
10:56:26 [dae782a29f5c7] waiting for daemon to start
10:56:26 [dae782a29f5c7] daemon started
10:56:26 
10:56:26 [defcdfc1ba008] waiting for daemon to start
10:56:26 [defcdfc1ba008] daemon started
10:56:26 
10:56:26 [dd4f748b5d563] waiting for daemon to start
10:56:26 [dd4f748b5d563] daemon started
10:56:26 
10:56:26 [defcdfc1ba008] exiting daemon
10:56:26 [defcdfc1ba008] waiting for daemon to start
10:56:26 [defcdfc1ba008] daemon started
10:56:26 
10:56:26 [defcdfc1ba008] exiting daemon
10:56:26 docker_cli_swarm_test.go:1160:
10:56:26     c.Assert(err, checker.IsNil, check.Commentf("out: %v", outs))
10:56:26 ... value *exec.ExitError = &exec.ExitError{ProcessState:(*os.ProcessState)(0xc420152000), Stderr:[]uint8(nil)} ("exit status 1")
10:56:26 ... out: Error response from daemon: rpc error: code = DeadlineExceeded desc = context deadline exceeded
10:56:26 
10:56:26 
10:56:26 [dae782a29f5c7] exiting daemon
10:56:26 [dd4f748b5d563] daemon started
10:56:26 Attempt #2: daemon is still running with pid 9371
10:56:26 Attempt #3: daemon is still running with pid 9371
10:56:26 Attempt #4: daemon is still running with pid 9371
10:56:26 [dd4f748b5d563] exiting daemon
10:56:26 ... Panic: Fixture has panicked (see related PANIC)
10:56:26 
10:56:26 ----------------------------------------------------------------------
```

**Comment by cpuguy83** [comment]
Haven't seen this windows failure before, but looks unrelated:

```
09:05:41 ----------------------------------------------------------------------
09:05:41 FAIL: docker_api_exec_test.go:159: DockerSuite.TestExecAPIStartValidCommand
09:05:41 
09:05:41 docker_api_exec_test.go:171:
09:05:41     c.Assert(inspectJSON.ExecIDs, checker.IsNil)
09:05:41 ... value []string = []string{"d626a340d4bbce20e177a7e4ed0d2a1081ae6451b5fd0ae1d47627982bbd5bc5"}
09:05:41 
09:05:45 
09:05:45 ----------------------------------------------------------------------
```

**Comment by cpuguy83** [comment]
Opened https://github.com/moby/moby/issues/34114 for the flakey test

**Comment by cpuguy83** [comment]
I think we're good on other issues.

**Comment by aaronlehmann** [comment]
All green, LGTM

File: integration-cli/fixtures/plugin/plugin_linux.go [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+	p, err := exec.LookPath(name)
+	if err == nil {
+		return p, nil
+	}
+
+	goBin, err := exec.LookPath("go")
+	if err != nil {
+		return "", err
+	}
+	installPath := filepath.Join(os.Getenv("GOPATH"), "bin", name)
```
Comment (AkihiroSuda) [inline integration-cli/fixtures/plugin/plugin_linux.go]: nit: No need to fix in this PR, but ideally we should support GOPATH with `:` separator.

**Review by AkihiroSuda — APPROVED** [review]
LGTM

cc @vdemeester

**Review by vdemeester — APPROVED** [review]
LGTM 👼

**Pull request merged**